backend/src/main/java/ai/verta/modeldb/lineage/LineageServiceImpl.java

```diff
@@ -299,33 +325,91 @@ private boolean filterLineageEntry(
 
   private void checkResourcesExistsAndAccessible(Session session, List<LineageEntry> lineageEntries)
       throws ModelDBException, NoSuchAlgorithmException, InvalidProtocolBufferException {
-    Set<String> experimentRuns = new HashSet<>();
+    Set<String> experimentRunIds = new HashSet<>();
     Map<Long, RepositoryContainer> blobs = new HashMap<>();
     for (LineageEntry lineageEntry : lineageEntries) {
-      validate(session, experimentRuns, blobs, lineageEntry);
+      validate(session, experimentRunIds, blobs, lineageEntry);
+    }
+    if (experimentRunIds.size() != 0) {
+      validate(session, experimentRunIds);
+    }
+  }
+
+  private Set<String> filter(Session session, Set<String> experimentRunIds) {
+    Set<String> result = new HashSet<>();
+    LinkedList<String> experimentRunIdsList = new LinkedList<>(experimentRunIds);
+    List<ExperimentRunEntity> experimentRunEntities =
+        experimentRunDAO.getExperimentRunEntitiesBatch(session, experimentRunIdsList);
+    for (ExperimentRunEntity experimentRunEntity : experimentRunEntities) {
+      if (experimentRunEntity == null) {
+        LOGGER.warn(EXPERIMENT_RUN_NOT_FOUND);
+      } else {
+        try {
+          roleService.validateEntityUserWithUserInfo(
+              ModelDBServiceResourceTypes.PROJECT,
+              experimentRunEntity.getProject_id(),
+              ModelDBServiceActions.READ);
+          result.add(experimentRunEntity.getId());
+        } catch (StatusRuntimeException e) {
+          LOGGER.warn("Can't access entity {}", e.getMessage());
```

Comment:
Is this expected to skip throwing an exception?

Reply:
yes